.github/workflows/security-scan.yml

```
@@ -0,0 +1,42 @@
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run npm audit
+        run: npm audit --production
+
+      - name: Run CodeQL Analysis
+        uses: github/codeql-action/init@v2
```

actionlint (1.7.7)

the runner of "github/codeql-action/init@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)